components/site/src/lib.rs

```diff
@@ -542,13 +547,19 @@ impl Site {
         let mut sass_options = Options::default();
         sass_options.output_style = OutputStyle::Compressed;
         for file in files {
-            let name = file.as_path().file_stem().unwrap().to_string_lossy();
-            let css = match compile_file(file.as_path(), sass_options.clone()) {
+            let css = match compile_file(&file, sass_options.clone()) {
                 Ok(c) => c,
                 Err(e) => bail!(e)
             };
 
-            create_file(&self.output_path.join(format!("{}.css", name)), &css)?;
+            let file_name = file.strip_prefix(&sass_path).unwrap().with_extension("css");
+            let might_need_dir_creation = file_name.parent().is_some();
+            let file_name = self.output_path.join(file_name);
+
+            if might_need_dir_creation {
```

Comment:
you can avoid creating the variable `might_need_dir_creation` and the `unwrap` below by using a `if let`

Reply:
There's a reason that this has been done this way, and I realize it's not obvious -- maybe you can help me figure out how to make it more clear.

* My intent is to get the `parent` BEFORE we `join` the `file_name` onto the `output_path` because the `output_path` ALWAYS gives this a parent -- it's nice to avoid system calls.
* However, when we actually call `create_all_dirs`, we want the `join`ed path, because otherwise we'd be missing the `public/` segment.

As an alternative approach, I could always just get the `parent` of the `join`ed path. That means that we'd be unconditionally calling the `create_all_dirs` function, though.

Reply:
I just tried editing this a bit to make the variable names more clear. What do you think?